### PR TITLE
Add teardown_module to test_queue.py

### PR DESCRIPTION
**Pull request by williamma12**
## What do these changes do?

Adds a `teardown_module` to `test_queue.py` in order to ensure that after `test_queue.py` finishes running, that there is no ray process still running.

### CODE REVIEW

File: python/ray/test/test_queue.py [hunk 1]
```diff
@@ -10,9 +10,18 @@
 from ray.experimental.queue import Queue, Empty, Full
 
 
-def setup_module():
-    if not ray.worker.global_worker.connected:
-        ray.init()
+@pytest.fixture
+def ray_start():
+    # Start the Ray process.
+    ray.init()
```
Comment (robertnishihara) [inline python/ray/test/test_queue.py]:
```suggestion
    ray.init(num_cpus=1)
```

File: python/ray/test/test_queue.py [hunk 2]
```diff
@@ -10,9 +10,18 @@
...
+    yield None
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
+
+
+def teardown_module():
```
Comment (robertnishihara) [inline python/ray/test/test_queue.py]: This should be unnecessary if we are passing `ray_start` in everywhere, right?

Comment (williamma12) [reply python/ray/test/test_queue.py]: Oops sorry I thought I took out `ray_start` since it did not work for `test_async` for some reason

Comment (robertnishihara) [reply python/ray/test/test_queue.py]: Does it actually not work with `test_async`? Any idea why that is?

Comment (williamma12) [reply python/ray/test/test_queue.py]: I'm not really sure, it just hangs on when we try to get the results `result = set(ray.get(consumers))` from the `Queue` https://github.com/ray-project/ray/pull/4012/files#diff-a97079c8e5770f39616578c37673e242R56

**Comment by robertnishihara** [comment]
I suspect the issue had to do with defining the remote functions outside of the tests. I pushed some small changes.

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**